plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,503 @@
+
+  ## specify server password
+  # password = "s#cr@t%"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = true
+`
+}
+
+func (r *RedisSentinel) Description() string {
+	return "Read metrics from one or many redis-sentinel servers"
+}
+
+func (r *RedisSentinel) Init() error {
+	if len(r.Servers) == 0 {
+		r.Servers = []string{"tcp://localhost:26379"}
+	}
+
+	r.clients = make([]*RedisSentinelClient, len(r.Servers))
+
+	tlsConfig, err := r.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	for i, serv := range r.Servers {
+		u, err := url.Parse(serv)
+		if err != nil {
+			return fmt.Errorf("unable to parse to address %q: %v", serv, err)
+		}
+
+		if u.Scheme != "tcp" && u.Scheme != "unix" {
+			return fmt.Errorf("invalid scheme %q. expected tcp or unix", u.Scheme)
+		}
+
+		password := r.Password
+
+		var address string
+		if u.Scheme == "unix" {
+			address = u.Path
+		} else {
+			address = u.Host
+		}
+
+		sentinel := redis.NewSentinelClient(
+			&redis.Options{
+				Addr:      address,
+				Password:  password,
+				Network:   u.Scheme,
+				PoolSize:  1,
+				TLSConfig: tlsConfig,
+			},
+		)
+
+		tags := map[string]string{}
+		if u.Scheme == "unix" {
+			tags["socket"] = u.Path
+		} else {
+			tags["source"] = u.Hostname()
+			tags["port"] = u.Port()
+		}
+
+		r.clients[i] = &RedisSentinelClient{
+			sentinel: sentinel,
+			tags:     tags,
+		}
+	}
+
+	return nil
+}
+
+// Redis list format has string key/values adjacent, so convert to a map for easier use
+func toMap(vals []interface{}) map[string]string {
+	m := make(map[string]string)
+
+	for idx := 0; idx < len(vals)-1; idx += 2 {
+		key, keyOk := vals[idx].(string)
+		value, valueOk := vals[idx+1].(string)
+
+		if keyOk && valueOk {
+			m[key] = value
+		}
+	}
+
+	return m
+}
+
+// Reads stats from all configured servers accumulates stats.
+// Returns one of the errors encountered while gather stats (if any).
+func (r *RedisSentinel) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, client := range r.clients {
+		wg.Add(1)
+
+		go func(acc telegraf.Accumulator, client *RedisSentinelClient) {
+			defer wg.Done()
+
+			gatherMasterStats(acc, client)
+			gatherInfoStats(acc, client)
+		}(acc, client)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+func gatherInfoStats(acc telegraf.Accumulator, client *RedisSentinelClient) {
+	infoCmd := redis.NewStringCmd("info", "all")
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(infoCmd)
+
+	info, infoErr := infoCmd.Result()
+	if infoErr != nil {
+		acc.AddError(infoErr)
+		return
+	}
+
+	rdr := strings.NewReader(info)
+	infoTags, infoFields, err := convertSentinelInfoOutput(client.baseTags(), rdr)
+	if err == nil {
+		acc.AddFields(measurementSentinel, infoFields, infoTags)
+	} else {
+		acc.AddError(err)
+	}
```

Comment:
Usually we check it the other way around
```suggestion
	if err != nil {
		acc.AddError(err)
		return
	}
	acc.AddFields(measurementSentinel, infoFields, infoTags)
```
So if you don't mind to change to this scheme it is easier to read for people familiar with telegraf.